### PR TITLE
[RFC] fix warnings with python 3.7

### PR DESCRIPTION
**Pull request by fansanelli**
Hello,

while trying to fix my docker containers I experienced these warning...
Can you please give it a try too?
Let's say this is not properly tested.

Francesco

### CODE REVIEW

**Comment by frodrigo** [comment]
No need to check for python 3 or 2, since python 2 support was removed.

File: requirements.txt [hunk 1]
```diff
@@ -7,4 +7,4 @@ mapbox-vector-tile
 pyclipper
 # Use a patch fork, for unmaintened rauth, https://github.com/litl/rauth/issues/185
 git+git://github.com/hendi/rauth.git@37615b738a944684bb02ea1c9ea5f3bbde84f0e7
-bottle == 0.12.13
+bottle == 0.12.18
```
Comment (jocelynj) [inline requirements.txt:10]: Does it belong in this commit?
There might be more trials to be done if we update bottle version.

Comment (fansanelli) [reply requirements.txt:10]: > Does it belong in this commit?
> There might be more trials to be done if we update bottle version.

It's intended, due to:

https://github.com/osm-fr/osmose-frontend/issues/248

This is not exactly TESTED, I'm trying to make my docker setup working in the meanwhile...

Comment (frodrigo) [reply requirements.txt:10]: Ok for getargspec to getfullargspec. Please just remove bottle == 0.12.18 from this PR.

Comment (fansanelli) [inline requirements.txt:10]:
```suggestion
bottle == 0.12.13
```

**Comment by fansanelli** [comment]
Hi,
in osmose_bottle.py

            prefix = route.config['mountpoint']['prefix']

can become

            prefix = route.config['mountpoint.prefix']

right?
Thanks
Francesco

**Comment by frodrigo** [comment]
> Hi,
> in osmose_bottle.py
> 
> ```
>         prefix = route.config['mountpoint']['prefix']
> ```
> 
> can become
> 
> ```
>         prefix = route.config['mountpoint.prefix']
> ```
> 
> right?
> Thanks
> Francesco

I don't know. The old one does not work any more?

**Comment by fansanelli** [comment]
It's another deprecation warning

**Comment by frodrigo** [comment]
I do not have this warning.

And what about the next line?
```
subapp = route.config['mountpoint']['target']
```

**Comment by fansanelli** [comment]
It's the same of course.. probability it's due to my Python version, but I can update the series with both of them changed if you agree.

**Comment by frodrigo** [comment]
But I want to be sure it have to be fixed.
I have no warning with fresh docker building and using bottle 2.12.18.

**Comment by fansanelli** [comment]
My setup is arch Linux with docker from AUR, I'm not sure if this matters.. Python version is 3.7.2

**Comment by frodrigo** [comment]
All your host setup does not matter if you use docker.

**Comment by fansanelli** [comment]
Right. But when I call the pages warnings are printed, or in case of bottle exception thrown

**Comment by frodrigo** [comment]
Do you have URL for this?

**Comment by fansanelli** [comment]
Tomorrow I will share a log. But when I start bottle exception is printed..

**Comment by fansanelli** [comment]
Hello @frodrigo,

I removed all the containers and patches as you can see from my log.
I followed the README blindly.

```
[francesco@hp-pavillion git]$ sudo docker container rm $(sudo docker container ls -aq)
d17490dee7c9
8db896c099cf
65fa9f3b9940
1945c4fb8657
[francesco@hp-pavillion git]$ sudo rm -rf osmose-frontend/
[sudo] password di francesco: 
[francesco@hp-pavillion git]$ git clone https://github.com/fansanelli/osmose-frontend.git
Clone in 'osmose-frontend' in corso...
remote: Enumerating objects: 64, done.
remote: Counting objects: 100% (64/64), done.
remote: Compressing objects: 100% (54/54), done.
remote: Total 12738 (delta 28), reused 24 (delta 10), pack-reused 12674
Ricezione degli oggetti: 100% (12738/12738), 8.27 MiB | 5.09 MiB/s, fatto.
Risoluzione dei delta: 100% (8141/8141), fatto.
[francesco@hp-pavillion git]$ cd osmose-frontend/
[francesco@hp-pavillion osmose-frontend]$ git remote add upstream https://github.com/osm-fr/osmose-frontend.git
[francesco@hp-pavillion osmose-frontend]$ git fetch upstream
remote: Enumerating objects: 18, done.
remote: Counting objects: 100% (18/18), done.
remote: Compressing objects: 100% (6/6), done.
remote: Total 13 (delta 9), reused 11 (delta 7), pack-reused 0
Decompressione degli oggetti in corso: 100% (13/13), 1.35 KiB | 230.00 KiB/s, fatto.
Da https://github.com/osm-fr/osmose-frontend
 * [nuovo branch]    cquest-patch-missing-data -> upstream/cquest-patch-missing-data
 * [nuovo branch]    jocelyn    -> upstream/jocelyn
 * [nuovo branch]    master     -> upstream/master
[francesco@hp-pavillion osmose-frontend]$ git pull upstream master
Da https://github.com/osm-fr/osmose-frontend
 * branch            master     -> FETCH_HEAD
Aggiornamento di 940faf8..638304f
Fast-forward
 docker/Dockerfile         | 5 ++---
 docker/README.md          | 1 +
 docker/docker-compose.yml | 3 +++
 3 files changed, 6 insertions(+), 3 deletions(-)
[francesco@hp-pavillion osmose-frontend]$ cd docker/
[francesco@hp-pavillion docker]$ curl http://osmose.openstreetmap.fr/export/osmose-menu.sql.bz2 | bzcat > osmose-menu.sql
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 35376  100 35376    0     0  23396      0  0:00:01  0:00:01 --:--:-- 23381
[francesco@hp-pavillion docker]$ sudo docker-compose build
postgres uses an image, skipping
Building frontend
Step 1/19 : FROM python:3.7
 ---> 072863679c31
Step 2/19 : MAINTAINER Frédéric Rodrigo <fred.rodrigo@gmail.com>
 ---> Using cache
 ---> ed55f96740f8
Step 3/19 : RUN mkdir -p /data/work/root/results/
 ---> Using cache
 ---> bdfb7b6b8ea7
Step 4/19 : RUN curl -sL http://deb.nodesource.com/setup_12.x -o nodesource_setup.sh &&     bash ./nodesource_setup.sh &&     apt update &&     apt install -y --no-install-recommends         bzip2         gettext         git         make         nodejs         postgresql-client         sudo &&     apt-get clean
 ---> Using cache
 ---> 620fe6fee05c
Step 5/19 : ADD ./requirements.txt /data/project/osmose/frontend/requirements.txt
 ---> Using cache
 ---> 165cc029eee8
Step 6/19 : WORKDIR /data/project/osmose/frontend
 ---> Using cache
 ---> 34177834856c
Step 7/19 : RUN pip install -r requirements.txt
 ---> Using cache
 ---> 8406919254b2
Step 8/19 : RUN mkdir /data/project/osmose/frontend/web
 ---> Using cache
 ---> ba7c9f7bac98
Step 9/19 : ADD ./web/package.json /data/project/osmose/frontend/web/
 ---> Using cache
 ---> 68430ee51893
Step 10/19 : ADD ./web/package-lock.json /data/project/osmose/frontend/web/
 ---> Using cache
 ---> e6675948518b
Step 11/19 : ADD ./web/webpack.config.js /data/project/osmose/frontend/web/
 ---> Using cache
 ---> ce5a4a748378
Step 12/19 : RUN mkdir ../node_modules &&     ln -s ../../node_modules web/node_modules &&     cd web &&     npm install
 ---> Using cache
 ---> 7e59d76cfd25
Step 13/19 : ADD ./web/static /data/project/osmose/frontend/web/static
 ---> 27593e74bc6b
Step 14/19 : RUN cd web && npm run build
 ---> Running in e629c4afabd6

> osmose-frontend@0.0.0 build /data/project/osmose/frontend/web
> webpack

(node:25) DeprecationWarning: Tapable.plugin is deprecated. Use new API on `.hooks` instead
Hash: dbd30d91f4ad5fe4eb10
Version: webpack 4.43.0
Time: 15553ms
Built at: 05/22/2020 3:41:13 PM
                                                        Asset      Size  Chunks                                Chunk Names
         ./static/images/aa734bf392c47c4c5c5a47be0cf47f3a.png  24.6 KiB          [emitted]                     
            static/map/webpack.bundle-dbd30d91f4ad5fe4eb10.js   710 KiB       1  [emitted] [immutable]  [big]  static/map
static/map/webpack.bundle-dbd30d91f4ad5fe4eb10.js.LICENSE.txt   2.2 KiB          [emitted]                     
        static/map/webpack.bundle-dbd30d91f4ad5fe4eb10.js.map  2.14 MiB       1  [emitted] [dev]               static/map
                static/webpack.bundle-dbd30d91f4ad5fe4eb10.js   336 KiB       0  [emitted] [immutable]  [big]  static
    static/webpack.bundle-dbd30d91f4ad5fe4eb10.js.LICENSE.txt  1.94 KiB          [emitted]                     
            static/webpack.bundle-dbd30d91f4ad5fe4eb10.js.map   997 KiB       0  [emitted] [dev]               static
Entrypoint static [big] = static/webpack.bundle-dbd30d91f4ad5fe4eb10.js static/webpack.bundle-dbd30d91f4ad5fe4eb10.js.map
Entrypoint static/map [big] = static/map/webpack.bundle-dbd30d91f4ad5fe4eb10.js static/map/webpack.bundle-dbd30d91f4ad5fe4eb10.js.map
 [6] ./static/map/layers.js + 1 modules 5.61 KiB {1} [built]
     | ./static/map/layers.js 3.53 KiB [built]
     | ./static/map/Mapillary.js 2.08 KiB [built]
[16] ./static/css/style.css 1.19 KiB {0} {1} [built]
[17] /data/project/osmose/node_modules/css-loader!/data/project/osmose/node_modules/sprite-loader??ref--5-2!./static/css/style.css 732 bytes {0} {1} [built]
[21] ./static/map/Osmose.Menu.js 8.12 KiB {1} [built]
[22] ./static/map/Osmose.Doc.js 3.96 KiB {1} [built]
[29] ./static/map/map.js 4.27 KiB {1} [built]
[30] ./static/map/Osmose.Export.js 845 bytes {1} [built]
[31] ./static/map/Osmose.Editor.js 10.2 KiB {1} [built]
[32] ./static/map/Osmose.Marker.js 8.29 KiB {1} [built]
[34] ./static/webpack.index.js 156 bytes {0} [built]
[35] ./static/js/err_delete.js 927 bytes {0} [built]
[36] ./static/js/sorttable.js 16.9 KiB {0} [built]
[37] ./static/map/webpack.index.js 415 bytes {1} [built]
[78] ./static/map/style.css 1.19 KiB {1} [built]
[80] ./static/images/markers/markers-l.css 1.22 KiB {1} [built]
    + 68 hidden modules

WARNING in asset size limit: The following asset(s) exceed the recommended size limit (244 KiB).
This can impact web performance.
Assets: 
  static/webpack.bundle-dbd30d91f4ad5fe4eb10.js (336 KiB)
  static/map/webpack.bundle-dbd30d91f4ad5fe4eb10.js (710 KiB)

WARNING in entrypoint size limit: The following entrypoint(s) combined asset size exceeds the recommended limit (244 KiB). This can impact web performance.
Entrypoints:
  static (336 KiB)
      static/webpack.bundle-dbd30d91f4ad5fe4eb10.js
  static/map (710 KiB)
      static/map/webpack.bundle-dbd30d91f4ad5fe4eb10.js


WARNING in webpack performance recommendations: 
You can limit the size of your bundles by using import() or require.ensure to lazy load some parts of your application.
For more info visit https://webpack.js.org/guides/code-splitting/
Removing intermediate container e629c4afabd6
 ---> ac8fc6017dcc
Step 15/19 : ADD . /data/project/osmose/frontend
 ---> 5121944a742b
Step 16/19 : RUN cd web/po &&     make mo &&     cd ..
 ---> Running in 9d0c3089fae3
msgfmt --output-file=mo/lt/LC_MESSAGES/osmose-frontend.mo lt.po
msgfmt --output-file=mo/ja/LC_MESSAGES/osmose-frontend.mo ja.po
msgfmt --output-file=mo/en/LC_MESSAGES/osmose-frontend.mo en.po
msgfmt --output-file=mo/fr/LC_MESSAGES/osmose-frontend.mo fr.po
msgfmt --output-file=mo/fa/LC_MESSAGES/osmose-frontend.mo fa.po
msgfmt --output-file=mo/vi/LC_MESSAGES/osmose-frontend.mo vi.po
msgfmt --output-file=mo/da/LC_MESSAGES/osmose-frontend.mo da.po
msgfmt --output-file=mo/ca/LC_MESSAGES/osmose-frontend.mo ca.po
msgfmt --output-file=mo/ru/LC_MESSAGES/osmose-frontend.mo ru.po
msgfmt --output-file=mo/ro/LC_MESSAGES/osmose-frontend.mo ro.po
msgfmt --output-file=mo/nb/LC_MESSAGES/osmose-frontend.mo nb.po
msgfmt --output-file=mo/de/LC_MESSAGES/osmose-frontend.mo de.po
msgfmt --output-file=mo/fi/LC_MESSAGES/osmose-frontend.mo fi.po
msgfmt --output-file=mo/es/LC_MESSAGES/osmose-frontend.mo es.po
msgfmt --output-file=mo/eu/LC_MESSAGES/osmose-frontend.mo eu.po
msgfmt --output-file=mo/pl/LC_MESSAGES/osmose-frontend.mo pl.po
msgfmt --output-file=mo/cs/LC_MESSAGES/osmose-frontend.mo cs.po
msgfmt --output-file=mo/hu/LC_MESSAGES/osmose-frontend.mo hu.po
msgfmt --output-file=mo/it/LC_MESSAGES/osmose-frontend.mo it.po
msgfmt --output-file=mo/gl/LC_MESSAGES/osmose-frontend.mo gl.po
msgfmt --output-file=mo/pt_BR/LC_MESSAGES/osmose-frontend.mo pt_BR.po
msgfmt --output-file=mo/uk/LC_MESSAGES/osmose-frontend.mo uk.po
msgfmt --output-file=mo/nl/LC_MESSAGES/osmose-frontend.mo nl.po
msgfmt --output-file=mo/sv/LC_MESSAGES/osmose-frontend.mo sv.po
msgfmt --output-file=mo/zh_CN/LC_MESSAGES/osmose-frontend.mo zh_CN.po
msgfmt --output-file=mo/pt/LC_MESSAGES/osmose-frontend.mo pt.po
msgfmt --output-file=mo/zh_TW/LC_MESSAGES/osmose-frontend.mo zh_TW.po
msgfmt --output-file=mo/el/LC_MESSAGES/osmose-frontend.mo el.po
Removing intermediate container 9d0c3089fae3
 ---> 34ee12f6fc06
Step 17/19 : ENV LANG en_US.UTF-8
 ---> Running in 0491c6d71ad2
Removing intermediate container 0491c6d71ad2
 ---> 121b9be058c0
Step 18/19 : CMD ./osmose-standalone-bottle.py
 ---> Running in ae0d00ad89d6
Removing intermediate container ae0d00ad89d6
 ---> a3f094246141
Step 19/19 : EXPOSE 20009
 ---> Running in 6f697575d3c9
Removing intermediate container 6f697575d3c9
 ---> a31ce7d511fc

Successfully built a31ce7d511fc
Successfully tagged osm-fr/osmose_frontend:latest
[francesco@hp-pavillion docker]$ sudo docker-compose -f docker-compose.yml -f docker-compose-test.yml -f docker-compose-dev.yml run --name frontend -p 20009:20009 frontend
Creating docker_postgres_1 ... done
root@6d44f5ae5917:/data/project/osmose/frontend# cd web
root@6d44f5ae5917:/data/project/osmose/frontend/web# cd po && make mo && cd ..
msgfmt --output-file=mo/lt/LC_MESSAGES/osmose-frontend.mo lt.po
msgfmt --output-file=mo/ja/LC_MESSAGES/osmose-frontend.mo ja.po
msgfmt --output-file=mo/en/LC_MESSAGES/osmose-frontend.mo en.po
msgfmt --output-file=mo/fr/LC_MESSAGES/osmose-frontend.mo fr.po
msgfmt --output-file=mo/fa/LC_MESSAGES/osmose-frontend.mo fa.po
msgfmt --output-file=mo/vi/LC_MESSAGES/osmose-frontend.mo vi.po
msgfmt --output-file=mo/da/LC_MESSAGES/osmose-frontend.mo da.po
msgfmt --output-file=mo/ca/LC_MESSAGES/osmose-frontend.mo ca.po
msgfmt --output-file=mo/ru/LC_MESSAGES/osmose-frontend.mo ru.po
msgfmt --output-file=mo/ro/LC_MESSAGES/osmose-frontend.mo ro.po
msgfmt --output-file=mo/nb/LC_MESSAGES/osmose-frontend.mo nb.po
msgfmt --output-file=mo/de/LC_MESSAGES/osmose-frontend.mo de.po
msgfmt --output-file=mo/fi/LC_MESSAGES/osmose-frontend.mo fi.po
msgfmt --output-file=mo/es/LC_MESSAGES/osmose-frontend.mo es.po
msgfmt --output-file=mo/eu/LC_MESSAGES/osmose-frontend.mo eu.po
msgfmt --output-file=mo/pl/LC_MESSAGES/osmose-frontend.mo pl.po
msgfmt --output-file=mo/cs/LC_MESSAGES/osmose-frontend.mo cs.po
msgfmt --output-file=mo/hu/LC_MESSAGES/osmose-frontend.mo hu.po
msgfmt --output-file=mo/it/LC_MESSAGES/osmose-frontend.mo it.po
msgfmt --output-file=mo/gl/LC_MESSAGES/osmose-frontend.mo gl.po
msgfmt --output-file=mo/pt_BR/LC_MESSAGES/osmose-frontend.mo pt_BR.po
msgfmt --output-file=mo/uk/LC_MESSAGES/osmose-frontend.mo uk.po
msgfmt --output-file=mo/nl/LC_MESSAGES/osmose-frontend.mo nl.po
msgfmt --output-file=mo/sv/LC_MESSAGES/osmose-frontend.mo sv.po
msgfmt --output-file=mo/zh_CN/LC_MESSAGES/osmose-frontend.mo zh_CN.po
msgfmt --output-file=mo/pt/LC_MESSAGES/osmose-frontend.mo pt.po
msgfmt --output-file=mo/zh_TW/LC_MESSAGES/osmose-frontend.mo zh_TW.po
msgfmt --output-file=mo/el/LC_MESSAGES/osmose-frontend.mo el.po
root@6d44f5ae5917:/data/project/osmose/frontend/web# ln -s ../../node_modules/ node_modules
root@6d44f5ae5917:/data/project/osmose/frontend/web# npm run build

> osmose-frontend@0.0.0 build /data/project/osmose/frontend/web
> webpack

(node:82) DeprecationWarning: Tapable.plugin is deprecated. Use new API on `.hooks` instead
Hash: dbd30d91f4ad5fe4eb10
Version: webpack 4.43.0
Time: 7552ms
Built at: 05/22/2020 3:43:14 PM
                                                        Asset      Size  Chunks                                Chunk Names
         ./static/images/aa734bf392c47c4c5c5a47be0cf47f3a.png  24.6 KiB          [emitted]                     
            static/map/webpack.bundle-dbd30d91f4ad5fe4eb10.js   710 KiB       1  [emitted] [immutable]  [big]  static/map
static/map/webpack.bundle-dbd30d91f4ad5fe4eb10.js.LICENSE.txt   2.2 KiB          [emitted]                     
        static/map/webpack.bundle-dbd30d91f4ad5fe4eb10.js.map  2.14 MiB       1  [emitted] [dev]               static/map
                static/webpack.bundle-dbd30d91f4ad5fe4eb10.js   336 KiB       0  [emitted] [immutable]  [big]  static
    static/webpack.bundle-dbd30d91f4ad5fe4eb10.js.LICENSE.txt  1.94 KiB          [emitted]                     
            static/webpack.bundle-dbd30d91f4ad5fe4eb10.js.map   997 KiB       0  [emitted] [dev]               static
Entrypoint static [big] = static/webpack.bundle-dbd30d91f4ad5fe4eb10.js static/webpack.bundle-dbd30d91f4ad5fe4eb10.js.map
Entrypoint static/map [big] = static/map/webpack.bundle-dbd30d91f4ad5fe4eb10.js static/map/webpack.bundle-dbd30d91f4ad5fe4eb10.js.map
 [6] ./static/map/layers.js + 1 modules 5.61 KiB {1} [built]
     | ./static/map/layers.js 3.53 KiB [built]
     | ./static/map/Mapillary.js 2.08 KiB [built]
[16] ./static/css/style.css 1.19 KiB {0} {1} [built]
[17] /data/project/osmose/node_modules/css-loader!/data/project/osmose/node_modules/sprite-loader??ref--5-2!./static/css/style.css 732 bytes {0} {1} [built]
[21] ./static/map/Osmose.Menu.js 8.12 KiB {1} [built]
[22] ./static/map/Osmose.Doc.js 3.96 KiB {1} [built]
[29] ./static/map/map.js 4.27 KiB {1} [built]
[30] ./static/map/Osmose.Export.js 845 bytes {1} [built]
[31] ./static/map/Osmose.Editor.js 10.2 KiB {1} [built]
[32] ./static/map/Osmose.Marker.js 8.29 KiB {1} [built]
[34] ./static/webpack.index.js 156 bytes {0} [built]
[35] ./static/js/err_delete.js 927 bytes {0} [built]
[36] ./static/js/sorttable.js 16.9 KiB {0} [built]
[37] ./static/map/webpack.index.js 415 bytes {1} [built]
[78] ./static/map/style.css 1.19 KiB {1} [built]
[80] ./static/images/markers/markers-l.css 1.22 KiB {1} [built]
    + 68 hidden modules

WARNING in asset size limit: The following asset(s) exceed the recommended size limit (244 KiB).
This can impact web performance.
Assets: 
  static/webpack.bundle-dbd30d91f4ad5fe4eb10.js (336 KiB)
  static/map/webpack.bundle-dbd30d91f4ad5fe4eb10.js (710 KiB)

WARNING in entrypoint size limit: The following entrypoint(s) combined asset size exceeds the recommended limit (244 KiB). This can impact web performance.
Entrypoints:
  static (336 KiB)
      static/webpack.bundle-dbd30d91f4ad5fe4eb10.js
  static/map (710 KiB)
      static/map/webpack.bundle-dbd30d91f4ad5fe4eb10.js


WARNING in webpack performance recommendations: 
You can limit the size of your bundles by using import() or require.ensure to lazy load some parts of your application.
For more info visit https://webpack.js.org/guides/code-splitting/
root@6d44f5ae5917:/data/project/osmose/frontend/web# cd ..
root@6d44f5ae5917:/data/project/osmose/frontend# ./osmose-standalone-bottle.py
Bottle v0.12.13 server starting up (using WSGIRefServer())...
Listening on http://0.0.0.0:20009/
Hit Ctrl-C to quit.

Exception in thread Thread-2:
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/threading.py", line 926, in _bootstrap_inner
    self.run()
  File "/usr/local/lib/python3.7/site-packages/bottle.py", line 3158, in run
    if path[-4:] in ('.pyo', '.pyc'): path = path[:-1]
TypeError: 'NoneType' object is not subscriptable
```
This goes away upgrading bottle.

Open:
http://localhost:20009/en/errors?item=xxxx&useDevItem=true
```
/data/project/osmose/frontend/modules/bottle_user.py:30: DeprecationWarning: inspect.getargspec() is deprecated since Python 3.0, use inspect.signature() or inspect.getfullargspec()
  args = inspect.getargspec(route.callback)[0]
/data/project/osmose/frontend/modules/bottle_gettext.py:67: DeprecationWarning: inspect.getargspec() is deprecated since Python 3.0, use inspect.signature() or inspect.getfullargspec()
  args = inspect.getargspec(route.callback)[0]
/data/project/osmose/frontend/modules/bottle_pgsql.py:88: DeprecationWarning: inspect.getargspec() is deprecated since Python 3.0, use inspect.signature() or inspect.getfullargspec()
  args = inspect.getargspec(route.callback)[0]
/usr/local/lib/python3.7/site-packages/bottle.py:3468: DeprecationWarning: Flags not at the start of the expression "((?m)[urbURB]?(?:''(" (truncated)
  patterns = [re.compile(p%pattern_vars) for p in patterns]
/usr/local/lib/python3.7/site-packages/bottle.py:3468: DeprecationWarning: Flags not at the start of the expression '\\{\\{((?:((?m)[urbURB' (truncated)
  patterns = [re.compile(p%pattern_vars) for p in patterns]
/usr/local/lib/python3.7/site-packages/bottle.py:3573: DeprecationWarning: The include and rebase keywords are functions now.
  line, comment = self.fix_backward_compatibility(line, comment)
172.20.0.1 - - [22/May/2020 15:45:12] "GET /en/errors?item=xxxx&useDevItem=true HTTP/1.1" 200 15649
172.20.0.1 - - [22/May/2020 15:45:12] "GET /en/webpack.bundle-dbd30d91f4ad5fe4eb10.js HTTP/1.1" 200 344148
```

Then open: http://localhost:20009/
Click on false positive.

```
172.20.0.1 - - [22/May/2020 15:48:44] "GET / HTTP/1.1" 303 0
172.20.0.1 - - [22/May/2020 15:48:44] "GET / HTTP/1.1" 303 0
172.20.0.1 - - [22/May/2020 15:48:44] "GET /it/ HTTP/1.1" 200 771
172.20.0.1 - - [22/May/2020 15:48:44] "GET /it/webpack.bundle-dbd30d91f4ad5fe4eb10.js HTTP/1.1" 304 0
Traceback (most recent call last):
  File "/data/project/osmose/frontend/modules/bottle_pgsql.py", line 108, in wrapper
    rv = callback(*args, **kwargs)
  File "/data/project/osmose/frontend/modules/bottle_cors.py", line 32, in wrapper
    return callback(*args, **kwargs)
  File "/data/project/osmose/frontend/modules/bottle_gettext.py", line 99, in wrapper
    return callback(*args, **kwargs)
  File "/data/project/osmose/frontend/web/issues.py", line 117, in index
    errors = query._gets(db, params)
  File "/data/project/osmose/frontend/modules/query.py", line 254, in _gets
    db.execute(sql) # FIXME pas de %
  File "/usr/local/lib/python3.7/site-packages/psycopg2/extras.py", line 143, in execute
    return super(DictCursor, self).execute(query, vars)
psycopg2.errors.UndefinedTable: missing FROM-clause entry for table "dynpoi_class"
LINE 22:             dynpoi_class.item = class.item AND
                     ^

None
172.20.0.1 - - [22/May/2020 15:48:47] "GET /it/errors/false-positive HTTP/1.1" 500 990
```
A wrong url like:
http://localhost:20009/it/xxx

will throw
```
/data/project/osmose/frontend/modules/bottle_user.py:30: DeprecationWarning: inspect.getargspec() is deprecated since Python 3.0, use inspect.signature() or inspect.getfullargspec()
  args = inspect.getargspec(route.callback)[0]
/data/project/osmose/frontend/modules/bottle_gettext.py:67: DeprecationWarning: inspect.getargspec() is deprecated since Python 3.0, use inspect.signature() or inspect.getfullargspec()
  args = inspect.getargspec(route.callback)[0]
/data/project/osmose/frontend/modules/bottle_pgsql.py:88: DeprecationWarning: inspect.getargspec() is deprecated since Python 3.0, use inspect.signature() or inspect.getfullargspec()
  args = inspect.getargspec(route.callback)[0]
/usr/local/lib/python3.7/site-packages/bottle.py:3468: DeprecationWarning: Flags not at the start of the expression "((?m)[urbURB]?(?:''(" (truncated)
  patterns = [re.compile(p%pattern_vars) for p in patterns]
/usr/local/lib/python3.7/site-packages/bottle.py:3468: DeprecationWarning: Flags not at the start of the expression '\\{\\{((?:((?m)[urbURB' (truncated)
  patterns = [re.compile(p%pattern_vars) for p in patterns]
/data/project/osmose/frontend/modules/osmose_bottle.py:44: DeprecationWarning: Accessing namespaces as dicts is discouraged. Only use flat item access: cfg["names"]["pace"]["key"] -> cfg["name.space.key"]
  prefix = route.config['mountpoint']['prefix']
/data/project/osmose/frontend/modules/osmose_bottle.py:45: DeprecationWarning: Accessing namespaces as dicts is discouraged. Only use flat item access: cfg["names"]["pace"]["key"] -> cfg["name.space.key"]
  subapp = route.config['mountpoint']['target']
/usr/local/lib/python3.7/site-packages/bottle.py:3573: DeprecationWarning: The include and rebase keywords are functions now.
  line, comment = self.fix_backward_compatibility(line, comment)
172.20.0.1 - - [22/May/2020 15:50:46] "GET /it/xxx HTTP/1.1" 404 13313
172.20.0.1 - - [22/May/2020 15:50:46] "GET /it/favicon.png HTTP/1.1" 200 135
```

**Comment by frodrigo** [comment]
Ok. Thank you.
Can add to this PR the fix for the 2 lines, then I will merge this part:
```
/data/project/osmose/frontend/modules/osmose_bottle.py:45: DeprecationWarning: Accessing namespaces as dicts is discouraged. Only use flat item access: cfg["names"]["pace"]["key"] -> cfg["name.space.key"]
```

**Comment by frodrigo** [comment]
Thank you.

Open new PR if you want to fix more.

**Comment by fansanelli** [comment]
Thanks!

**Pull request merged**